Objects/tupleobject.c

```diff
@@ -119,7 +113,34 @@ PyTuple_New(Py_ssize_t size)
         if (op == NULL)
             return NULL;
     }
-    for (i=0; i < size; i++)
+    return (PyObject *) op;
+}
+
+_Py_NO_INLINE static PyObject *
+tuple_new_untracked(Py_ssize_t size)
+{
+    assert(size != 0);
```

Comment:
I suggest "assert(size > 0);". It doesn't sound like a good idea to allocate a tuple with a negative size.

nitpick: remove the empty line :-)

Reply:
`assert(size > 0)` will cause `PyTuple_Pack(-1)` to fail assertion instead of raising `PyErr_BadInternalCall()`, is that OK?

Reply:
Would would you call PyTuple_Pack(-1)? :-) What's the current behavior?